lib/identity/stateTransitions/assetLock/proof/chain/validateChainAssetLockProofStructureFactory.js

```diff
@@ -0,0 +1,127 @@
+const { Transaction } = require('@dashevo/dashcore-lib');
+const chainAssetLockProofSchema = require('../../../../../../schema/identity/stateTransition/assetLock/proof/chainAssetLockProof.json');
+
+const convertBuffersToArrays = require('../../../../../util/convertBuffersToArrays');
+const InvalidIdentityAssetLockProofHeightError = require('../../../../../errors/InvalidIdentityAssetLockProofHeightError');
+const InvalidIdentityAssetLockProofOutPointError = require('../../../../../errors/InvalidIdentityAssetLockProofOutPointError');
+const InvalidIdentityAssetLockTransactionError = require('../../../../../errors/InvalidIdentityAssetLockTransactionError');
+const IdentityAssetLockTransactionOutputNotFoundError = require('../../../../../errors/IdentityAssetLockTransactionOutputNotFoundError');
+const InvalidIdentityAssetLockTransactionOutputError = require('../../../../../errors/InvalidIdentityAssetLockTransactionOutputError');
+const InvalidIdentityAssetLockProofOutPointTransactionError = require('../../../../../errors/InvalidIdentityAssetLockProofOutPointTransactionError');
+
+/**
+ * @param {JsonSchemaValidator} jsonSchemaValidator
+ * @param {StateRepository} stateRepository
+ * @returns {validateChainAssetLockProofStructure}
+ */
+function validateChainAssetLockProofStructureFactory(
+  jsonSchemaValidator,
+  stateRepository,
+) {
+  /**
+   * @typedef {validateChainAssetLockProofStructure}
+   * @param {RawChainAssetLockProof} rawAssetLockProof
+   */
+  async function validateChainAssetLockProofStructure(
+    rawAssetLockProof,
+  ) {
+    const result = jsonSchemaValidator.validate(
+      chainAssetLockProofSchema,
+      convertBuffersToArrays(rawAssetLockProof),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    const {
+      coreChainLockedHeight: proofCoreChainLockedHeight,
+      outPoint: outPointBuffer,
+    } = rawAssetLockProof;
+
+    const latestPlatformBlockHeader = await stateRepository.fetchLatestPlatformBlockHeader();
+
+    const { coreChainLockedHeight: currentCoreChainLockedHeight } = latestPlatformBlockHeader;
+
+    if (currentCoreChainLockedHeight > proofCoreChainLockedHeight) {
+      result.addError(
+        new InvalidIdentityAssetLockProofHeightError(
+          proofCoreChainLockedHeight,
+          currentCoreChainLockedHeight,
+        ),
+      );
+
+      return result;
+    }
+
+    if (!await stateRepository.isAssetLockTransactionOutPointNotUsed(outPointBuffer)) {
```

Comment:
Extract code since 57 to 119 to `validateAssetLockTrasaction` and use here and in `validateInstantAssetLockProofStructure`